src/components/StatusWidget.tsx

```diff
@@ -32,21 +39,57 @@ export class StatusWidget extends ReactWidget {
     }
   }
 
+  /**
+   * Boolean indicating whether credentials are required.
+   */
+  get waitingForCredentials(): boolean {
+    return this._waitingForCredentials;
+  }
+
+  /**
+   * Sets the boolean indicating whether credentials are required.
+   */
+  set waitingForCredentials(value: boolean) {
+    this._waitingForCredentials = value;
+  }
+
```

Comment:
We will use the model signal directly - the flaw of the current code is that we don't update the widget when setting the value (it seems there is parasite update request that are giving a wrong impression of live udpate).

```suggestion
```